### PR TITLE
feat(LIB-1360): add Label on FzSelect

### PR DESCRIPTION
**Pull request by Crish15**
Jira: [LIB-1360](https://fiscozen.atlassian.net/browse/LIB-1360)

### CODE REVIEW

File: packages/select/src/FzSelect.vue [hunk 1]
```diff
@@ -67,10 +67,18 @@
       ref="containerRef"
       test-id="fzselect-options-container"
     >
-      <template v-if="visibleOptions.length">
+      <template
+        v-if="visibleOptions.length"
+        v-for="option in visibleOptions"
```
Comment (oxcened) [inline packages/select/src/FzSelect.vue:72]: anti-pattern to mix v-if and v-for

File: packages/select/src/types.ts [hunk 2]
```diff
@@ -77,4 +79,13 @@ export type FzSelectOptionsProps = {
...
   readonly?: boolean;
+  /**
+   * the type of option
+   */
+  kind?: "option";
+};
+
+export type FzSelectLabelProps = {
+  label: string;
+  kind: "label";
```
Comment (oxcened) [inline packages/select/src/types.ts:90]: missing documentation

File: packages/select/src/components/FzSelectLabel.vue [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { computed } from "vue";
+import { FzSelectLabelProps } from "../types";
+
+const props = defineProps<{
+  option: FzSelectLabelProps;
+  size: "sm" | "md" | "lg";
+}>();
+
+const staticClass =
+  "text-grey-400 flex items-center text-ellipsis whitespace-nowrap";
```
Comment (oxcened) [inline packages/select/src/components/FzSelectLabel.vue:22]: is it useful to have this declared on its own?

File: packages/select/src/components/FzSelectLabel.vue [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<template>
+  <button
+    disabled
+    test-id="fzselect-label"
+    :title="option.label"
+    :class="computedClass"
+  >
+    {{ option.label }}
+  </button>
```
Comment (oxcened) [inline packages/select/src/components/FzSelectLabel.vue:9]: why a button element for a label?

File: packages/select/src/components/FzSelectLabel.vue (hunk 3 above)
Comment (Crish15) [reply packages/select/src/components/FzSelectLabel.vue:22]: is cleaner and easier to move into a composable in the future (it share the same style as the action list label)

File: packages/select/src/components/FzSelectLabel.vue (hunk 4 above)
Comment (Crish15) [reply packages/select/src/components/FzSelectLabel.vue:9]: to interrupt the click event. Label or span elements with the disabled attribute do not work. I also tried blocking the click event with some ‘@click.stop’/‘@click.prevent’ modifiers, but they did nothing.

**Pull request merged**